relayer/relays/beacon/cache/cache.go

```diff
@@ -35,6 +68,89 @@ func (b *BeaconCache) SetLastSyncedSyncCommitteePeriod(period uint64) {
 	}
 }
 
+func (b *BeaconCache) AddCheckPoint(finalizedHeaderRoot common.Hash, blockRootsTree *ssz.Node, slot uint64) {
+	b.addSlot(slot)
+	b.Finalized.Checkpoints.Proofs[slot] = Proof{
+		FinalizedBlockRoot: finalizedHeaderRoot,
+		BlockRootsTree:     blockRootsTree,
+		Slot:               slot,
+	}
+
+	b.pruneOldCheckpoints()
+}
+
+func (b *BeaconCache) AddCheckPointSlots(slots []uint64) {
+	for _, slot := range slots {
+		b.addSlot(slot)
+	}
+}
+
+func (b *BeaconCache) GetClosestCheckpoint(slot uint64) (Proof, error) {
+	checkpointSlot, err := b.calculateClosestCheckpointSlot(slot)
+	if err != nil {
+		return Proof{}, err
+	}
```

Comment:
So what we will do to handle here? is that the long range attack you mentioned before and we need to manuallay or from governance call to add some ancestry checkpoints into beacon light client?